Character.cpp

```diff
@@ -98,20 +95,37 @@ void Character::attackInternal(Character& other)
             b) your stats are boosted 10%
             c) the initial value of your stats is updated to reflect this boosted stat for the next time you defeat another character.
       */
-        assert(false);
+        if(hitPoints < *initialHitPoints){
+            hitPoints = *initialHitPoints;
+        }
+        hitPoints *= 1.1f;
+        *initialHitPoints = hitPoints;
+
+        if(armor < *initialArmorLevel){
+            armor = *initialArmorLevel;
+        }
+        armor *= 1.1f;
+        *initialArmorLevel = armor;
+
+        if(attackDamage < *initialAttackDamage){
+            attackDamage = *initialAttackDamage;
+        }
+        attackDamage *= 1.1f;
+        *initialAttackDamage = attackDamage;
+        
```

Comment:
D.R.Y.
you're doing the same thing with 2 pairs of variables, 3 times over.
write a free function and call it 3x.
Also: curly braces go on their own line with no other code in this course.
- [ ] revise